transform-plugins/src/main/java/co/cask/hydrator/plugin/ValueMapper.java

```diff
@@ -0,0 +1,251 @@
+          defaultsMapping.put(defaultsArray[0], defaultsArray[1]);
+        }
+      }
+      String[] mappingArray = this.mapping.split(",");
+      for (String mapping : mappingArray) {
+        String[] mappingValueArray = mapping.split(":");
+        ValueMapping valueMapping = new ValueMapping();
+        valueMapping.setTargetField(mappingValueArray[2]);
+        valueMapping.setLookupTableName(mappingValueArray[1]);
+
+        if (defaultsMapping.containsKey(mappingValueArray[0])) {
+          valueMapping.setDefaultValue(defaultsMapping.get(mappingValueArray[0]));
+        }
+        mappingValues.put(mappingValueArray[0], valueMapping);
+      }
+    }
+  }
+
+  /**
+   * Creates output schema with the help of input schema and mapping
+   */
+  private Schema getOutputSchema(Schema inputSchema) throws IllegalArgumentException {
+    Schema outputSchema = schemaCache.get(inputSchema);
+    if (outputSchema != null) {
+      return outputSchema;
+    }
+
+    List<Schema.Field> outputFields = new ArrayList<>();
+    for (Schema.Field inputField : inputSchema.getFields()) {
+      if (mappingValues.containsKey(inputField.getName())) {
+        if (inputField.getSchema().getType() != Schema.Type.STRING && !inputField.getSchema().isNullable()) {
+          throw new IllegalArgumentException("Input field " + inputField.getName() + " type should be String");
+        } else {
+          outputFields.add(Schema.Field.of(mappingValues.get(inputField.getName()).getTargetField(),
+                                           inputField.getSchema()));
+          }
+      } else {
+        outputFields.add(inputField);
+      }
+    }
+    outputSchema = Schema.recordOf(inputSchema.getRecordName() + ".formatted", outputFields);
+    schemaCache.put(inputSchema, outputSchema);
+    return outputSchema;
+  }
+
+  /**
+   * retrieve lookup table from table name
+   */
+  private void setLookup(TransformContext context) {
+    for (String key :mappingValues.keySet()) {
+      ValueMapping mapping = mappingValues.get(key);
+      LookupTableConfig tableConfig = new LookupTableConfig(LookupTableConfig.TableType.DATASET);
+      DatasetProperties arguments = DatasetProperties.builder().addAll(tableConfig.getDatasetProperties()).build();
+      Lookup<String> lookupTable = context.provide(mapping.getLookupTableName(), arguments.getProperties());
+      mapping.setLookupTable(lookupTable);
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter) throws Exception {
+
+    StructuredRecord.Builder builder = StructuredRecord.builder(getOutputSchema(input.getSchema()));
+
+    for (Schema.Field sourceField : input.getSchema().getFields()) {
+      if (mappingValues.containsKey(sourceField.getName())) {
+        ValueMapping mapping = mappingValues.get(sourceField.getName());
+        if (input.get(sourceField.getName()) == null || input.get(sourceField.getName()).toString().isEmpty()) {
```

Comment:
since you already checked in getOutputSchema that this field is a string, you can just do:

```
String sourceVal = input.get(sourceField);
if (sourceVal == null || sourceVal.isEmpty()) {
...
}
```

In other words, there is no need to do `.toString()`.